test/test_dataflow.py

```diff
@@ -1,70 +1,69 @@
 from apache_beam.options.pipeline_options import PipelineOptions, GoogleCloudOptions
 from apache_beam.runners.portability.fn_api_runner.fn_runner import RunnerResult
 from apache_beam.testing.test_pipeline import TestPipeline
-from collections import defaultdict
+from collections import defaultdict, Counter
 from bigflow import JobContext, Workflow
+from bigflow import dataflow
 
 from bigflow.dataflow import BeamJob
 from bigflow.workflow import DEFAULT_EXECUTION_TIMEOUT_IN_SECONDS, DEFAULT_PIPELINE_LEVEL_EXECUTION_TIMEOUT_SHIFT_IN_SECONDS
 
 
-class CountWordsFn(beam.DoFn):
-    def __init__(self, CounterState):
-        super().__init__()
-        self.CounterState = CounterState
-
-    def process(self, element, *args, **kwargs):
-        word, count = element
-        self.CounterState.counter[word] += len(count)
-        yield word, len(count)
-
-
-class CounterState(object):
-    counter = defaultdict(int)
-
-
-class CounterStater(beam.PTransform):
-    def __init__(self, CounterState):
-        super().__init__()
-        self.CounterState = CounterState
-
-    def expand(self, records_to_delete):
-        return records_to_delete \
-               | "CounterStateCountedWords" >> beam.ParDo(
-            CountWordsFn(self.CounterState))
+class CountWordsDriver:
 
+    counter = Counter()
 
-class CountWordsDriver:
-    def __init__(self, CounterStater):
-        self.result = None
+    def __init__(self):
         self.context = None
         self.pipeline = None
-        self.CounterStater = CounterStater
+        self.counter.clear()
 
-    def run(self, pipeline: Pipeline, context: JobContext, driver_arguments: dict):
-        words_input = pipeline | 'LoadingWordsInput' >> beam.Create(driver_arguments['words_to_count'])
-        words_input | 'FilterWords' >> (beam.Filter(lambda w: w in driver_arguments['words_to_filter'])
-                                        | 'MapToCount' >> beam.Map(lambda w: (w, 1))
-                                        | 'GroupWords' >> beam.GroupByKey()
-                                        | 'CountWords' >> self.CounterStater)
+    @staticmethod
+    def count(word, counts):
+        CountWordsDriver.counter[word] += sum(counts)
+
+    def run_ex(self, pipeline, context, words_to_count, words_to_filter):
+        self.pipeline = pipeline
         self.context = context
+        return (
+            pipeline
+            | 'LoadingWordsInput' >> beam.Create(words_to_count)
+            | 'FilterWords'       >> beam.Filter(lambda w: w in words_to_filter)
+            | 'MapToCount'        >> beam.Map(lambda w: (w, 1))
+            | 'GroupWords'        >> beam.GroupByKey()
+            | 'Count'             >> beam.MapTuple(self.count)
+        )
+
+    def run(self, pipeline: Pipeline, context: JobContext, driver_arguments: dict):
+        return self.run_ex(pipeline, context, driver_arguments['words_to_count'], driver_arguments['words_to_filter'])
+
+    def nope(self, pipeline, context):
```

Comment:
nope?

Reply:
Yes. Sort of "do nothing" (because for most cases it is ok to have empty pipline).

Reply:
So maybe rename it to do_nothing?